core/state/statedb.go

```diff
@@ -463,8 +463,6 @@ func (s *StateDB) deleteStateObject(obj *stateObject) {
 	}
 	// Delete the account from the trie
 	addr := obj.Address()
-	// Get absense proof of account in case the deletion needs the sister node.
-	oracle.PrefetchAccount(big.NewInt(s.db.BlockNumber.Int64()+1), addr, trie.GenPossibleShortNodePreimage)
```

Comment:
This is required. The reason being that after all state updates are done, if an account is empty (all fields zero) then EIP158 mandates that it should be removed from the trie, which happens here.

A proof (= Merkle trie path) for the account has not been fetched before, only proofs against the storage root of the accounts.

However — the GenPossibleShortNodePreImage call should be unecessary (given the new node cache) and that parameter can be replaced by `nil`.